elevation_mapping:
  ros__parameters:
    # frame IDs
    map_frame: "map"
    robot_frame: "base_link"
    # pose update
    use_pose_update: false
    pose_cache_size: 10
    # visibility clean up
    use_visibility_clean_up: false
    visibility_clean_up_duration: 1.0
    # extract_vaild_area: false
    use_and_publish_fused_map: false


    elevation_map:
      logger_name: "ElevationMap"
      lateral_length: 10.0
      longitudinal_length: 10.0
      resolution: 0.05
      initial_position_x: 0.0
      initial_position_y: 0.0
      min_normal_variance: 0.0001
      max_normal_variance: 0.05
      min_horizontal_variance: 0.0025
      max_horizontal_variance: 0.5
      mahalanobis_distance_thres: 2.5
      multi_height_noise: 0.00009
      scanning_duration: 0.5
      increase_height_alpha: 0.1

    # sensor processing
    sensors:
      sensor1:
        sensor_type: "perfect"
        sensor_frame: "perfect_camera_link"
        logger_name: PerfectSensorProcessor1
        use_voxel_filter: true
        voxel_leaf_size: 0.05
        pass_filter_lower_threshold: -100.0 # -infinity
        pass_filter_upper_threshold: 2.0

      sensor2:
        sensor_type: "stereo"
        sensor_frame: "stereo_camera_link"
        logger_name: StereoSensorProcessor2
        use_voxel_filter: true
        voxel_leaf_size: 0.05
        pass_filter_lower_threshold: -100.0 # -infinity
        pass_filter_upper_threshold: 2.0
        depth_upper_limit: 20.0
        depth_lower_limit: 0.1
        coef_normal_variance: 0.0001
        coef_lateral_variance: 0.0001

      sensor3:
        sensor_type: "laser"
        sensor_frame: "laser_link"
        logger_name: LaserSensorProcessor3
        use_voxel_filter: true
        voxel_leaf_size: 0.05
        pass_filter_lower_threshold: -100.0 # -infinity
        pass_filter_upper_threshold: 2.0
        laser_min_radius: 0.02
        laser_beam_angle: 0.0006
        laser_beam_constant: 0.0015

    robot_motion_updater:
      logger_name: RobotMotionUpdater